extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisStats.tsx

```diff
@@ -48,16 +48,16 @@ export const VariantAnalysisStats = ({
       return 'Failed';
     }
 
-    if (queryResult === 'warning') {
-      return 'Succeeded warnings';
+    if (variantAnalysisStatus === VariantAnalysisStatus.Canceled) {
+      return 'Stopped';
     }
 
-    if (queryResult === 'stopped') {
-      return 'Stopped';
+    if (variantAnalysisStatus === VariantAnalysisStatus.Succeeded && hasWarnings) {
+      return 'Succeeded warnings';
     }
 
     return 'Succeeded';
-  }, [variantAnalysisStatus, queryResult]);
+  }, [variantAnalysisStatus]);
```

Comment:
Do we need to include `hasWarnings` in our dependencies here?

Reply:
Yes, thanks for spotting that! Perhaps we should add [eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks) so we can catch these errors automatically in the future? My only worry is that it might give some linting errors on existing components.